packages/design/src/DataTable/Table.tsx

```diff
@@ -56,30 +59,51 @@ export function Table<T>({
   const renderBody = (data: T[]) => {
     const rows = [];
 
-    data.map((item, rowIdx) => {
-      const cells = columns.map((column, columnIdx) => {
-        const $cell = column.render ? (
-          column.render(item)
-        ) : (
-          <TextCell data={item[column.key]} />
-        );
+    if (fetching?.fetchStatus === 'loading') {
+      return <LoadingIndicator colSpan={columns.length} />;
+    } else {
+      data.map((item, rowIdx) => {
+        const cells = columns.map((column, columnIdx) => {
+          const $cell = column.render ? (
+            column.render(item)
+          ) : (
+            <TextCell data={item[column.key]} />
+          );
 
-        return (
-          <React.Fragment key={`${rowIdx} ${columnIdx}`}>
-            {$cell}
-          </React.Fragment>
-        );
+          return (
+            <React.Fragment key={`${rowIdx} ${columnIdx}`}>
+              {$cell}
+            </React.Fragment>
+          );
+        });
+        rows.push(<tr key={rowIdx}>{cells}</tr>);
       });
-      rows.push(<tr key={rowIdx}>{cells}</tr>);
-    });
 
-    if (rows.length) {
-      return <tbody>{rows}</tbody>;
-    }
+      if (rows.length) {
+        return <tbody>{rows}</tbody>;
+      }
 
-    return <EmptyIndicator emptyText={emptyText} colSpan={columns.length} />;
+      return <EmptyIndicator emptyText={emptyText} colSpan={columns.length} />;
+    }
   };
 
+  if (serverside) {
+    return (
+      <ServersideTable
+        style={style}
+        className={className}
+        data={state.data}
+        renderHeaders={renderHeaders}
+        renderBody={renderBody}
+        nextPage={nextPage}
+        prevPage={prevPage}
+        pagination={state.pagination}
+        fetching={fetching}
+        serverside={serverside}
+      />
+    );
+  }
+
   if (state.pagination) {
     return (
       <PagedTable
```

Comment:
Do we need two separate table components? The props look almost identical.

Reply:
The `PagedTable` uses the built-in `Pager` component while `ServersideTable` uses the `serversideSearchPanel` that is passed in. Even though they share many props, the `PagedTable` component already has a good amount of logic, and trying to merge the two tables into one component with more logic within the component to determine what to render would make it hard to read and follow. We tried a similar approach before (merging the basic, paginated, and searchable tables into one), but Alexey and I agreed that it was better to create a separate component for each type of Table. Let me know what you think

Reply:
Fair, let's not do anything too disruptive then.